### PR TITLE
Platform agnostic refactoring & cli testing

### PR DESCRIPTION
**Pull request by nothead31**
## Refactoring:

Refactored the build process to be more in line with PEP 517 (moving towards the new standard of using setup.cfg exclusively.)

Remaining issues:
- Some minor syntax and grammar corrections
- A few TODOs with notes on whether things should be kept or removed for clarity
- The setup.py script is still required by Travis, so instead of removing it, moved the static metadata to setup.cfg and kept the actual setup options in setup.py.

## CLI tests:

Added two CLI tests: one to test translations work and a second to test the API error. Additional tests will be added, but each requires some criteria:

- How many translation tests should be done through the CLI? These are better suited to the individual translation modules, the CLI functionality works. I could add tests for each service to ensure its being read properly, but still can't do any tests requiring an API.
- The -lang switch is broken. I'll be adding a bug report and new branch to fix that separately, since it doesn't apply to this mandate. That will require adding the get_languages functionality as a subcommand rather than as an option.
- Are there other functionality tests that I missed?

### CODE REVIEW

File: deep_translator/utils.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
 """
 utilities
 """
+# TODO: Discussion: Whats the purpsoe of this module?
```
Comment (nidhaloff) [inline deep_translator/utils.py:4]: we can remove utils.py. When I started the project I thought I can write all utilities functions here but looks like it is not needed

File: deep_translator/detection.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 language detection API
 """
 import requests
-from deep_translator.configs import config
+from .configs import config # TODO: Discussion: Could this be moved here and remove configs.py entirely?
```
Comment (nidhaloff) [inline deep_translator/detection.py:5]: we can remove configs.py

**Review by nidhaloff — APPROVED** [review]
Great work. I checked some of the changes and it looks good to me ;) 

I like the tests for the CLI. That was also the reason to write the CLI in click. It is easier to test a click cli app.

One thing I noticed is that you added a pyproject.toml. We can switch to working with pyproject.toml instead of setup.py but that will require some changes for travis ci I think. Or we can consider poetry (https://python-poetry.org/) for all dependency management and hence remove setup.py and setup.cfg

File: deep_translator/utils.py (hunk 1 above)
Comment (nothead31) [reply deep_translator/utils.py:4]: Removed

File: deep_translator/detection.py (hunk 2 above)
Comment (nothead31) [reply deep_translator/detection.py:5]: removed and config variable set up in detection.py, which is the only module that calls it.

**Pull request merged**